logstash-core/lib/logstash/plugin.rb

```diff
@@ -122,12 +123,22 @@ def metric
                          LogStash::Instrument::NamespacedNullMetric.new(@metric, :null)
                        end
   end
+
   # return the configured name of this plugin
   # @return [String] The name of the plugin defined by `config_name`
   def config_name
     self.class.config_name
   end
 
+  # commit event to dlq
+  def dlq_commit(event, reason)
+    if @dlq_manager.nil?
+      raise ArgumentError, "DLQ is not enabled, plugin cannot commit to it"
```

Comment:
I wonder ... how should plugin authors handle this error? What action should be taken if I try to dlq something and this ArgumentError happens?